### DISCUSSION TITLE
Interactive OAuth Authentication using azure AD

### DISCUSSION THREAD

**Discussion by Trivedi-Ashish** [dotnet/SqlClient#1589] [Q&A]
I have an ASP.Net MVC application in which i am using sqlclient.
For sql server authentication, I am planning to use interactive auth flow, either through browser popup/redirect or by device code.
I will be setting authorization property in the connection string appropriately. 
But what I am not sure how sqlclient will invoke interactive popup on the clients browser because interactive auth flow as far as i know is for client side applications but in my case I have an asp net mvc app which is a server side application.

Please can somebody help me out here?

**Comment by johnnypham** [comment]
I tried making an ASP.NET app that uses interactive authentication and a browser popup appeared. I assume this is what the client would experience too. Have you tried it yet?

**Reply by Trivedi-Ashish** [reply]
@johnnypham 
Did you deploy your application on IIS?
If I execute the application from visual studio IIS express, then popup appears. But if I publish the app to IIS, popup doesn’t appear.

**Comment by JRahnama** [comment]
@Trivedi-Ashish 
This is the official documentation on deploying IIS web apps with MFA authentication: [Configure Azure Multi-Factor Authentication Server for IIS web apps](https://docs.microsoft.com/en-us/azure/active-directory/authentication/howto-mfaserver-iis). If I have understood correct, you need to provide login url and other related stuff to IIS as well. Let me know if this could solve the issue for you.
